addon.py

```
@@ -139,38 +188,72 @@ def _handle_client(self, client):
+                                # Partial UTF-8 sequence at the buffer end, wait for more
+                                break
+                            
+                            print(f"Invalid UTF-8 in command stream: {str(exc)}")
+                            try:
+                                error_response = {"status": "error", "message": "Invalid UTF-8 in command stream"}
+                                client.sendall(json.dumps(error_response).encode('utf-8'))
+                            except socket.error as se:
+                                print(f"Failed to send UTF8 error response: {str(se)}")
+                            return
+                            
+                except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)